core/src/main/scala/scalaz/syntax/std/Function1Ops.scala

```diff
@@ -16,7 +16,7 @@ final class Function1Ops[T, R](private val self: T => R) extends AnyVal {
 
   def unary_!(implicit m: Memo[T, R]): T => R = m(self)
 
-  def toValidation[E](e: => E)(implicit ev: R =:= Boolean): T => Validation[NonEmptyList[E], T] =
+  def toValidation[E](e: => E)(implicit ev: R === Boolean): T => Validation[NonEmptyList[E], T] =
     (t: T) => (if (self(t): Boolean) Success(t) else Failure(NonEmptyList(e)))
 
```

Comment:
Once `R` really is understood to be `Boolean` by the copmiler, I'm assuming `(self(t): Boolean)` can be reduced to `self(t)`.

Reply:
Is there an implicit conversion when `R === Boolean` is in scope? Otherwise you need to do `ev(self(t))`.